java/rocksjni/writebatchhandlerjnicallback.cc

```diff
@@ -83,6 +88,92 @@ bool WriteBatchHandlerJniCallback::Continue() {
   return static_cast<bool>(jContinue == JNI_TRUE);
 }
 
+void WriteBatchHandlerJniCallback::SingleDelete(const Slice& key) {
+  const jbyteArray j_key = sliceToJArray(key);
+
+  m_env->CallVoidMethod(
+                        m_jWriteBatchHandler,
+                        m_jSingleDeleteMethodId,
+                        j_key);
+
+  m_env->DeleteLocalRef(j_key);
+}
+
+Status WriteBatchHandlerJniCallback::PutCF(uint32_t column_family_id, const Slice& key,
+                                             const Slice& value) {
+  if (column_family_id == 0) {
+    Put(key, value);
+  } else {
+    const jbyteArray j_key = sliceToJArray(key);
+    const jbyteArray j_value = sliceToJArray(value);
+
+    m_env->CallVoidMethod(
+                          m_jWriteBatchHandler,
+                          m_jPutCFMethodId,
+                          column_family_id,
+                          j_key,
+                          j_value);
+
+    m_env->DeleteLocalRef(j_value);
+    m_env->DeleteLocalRef(j_key);
+  }
+  return Status::OK();
```

Comment:
I think the status should be fed back from the Java API rather than just returning `Status::OK()`

Reply:
@min-mwei Okay, IMHO  the Java function signatures for `WriteBatch#Handler#put`should be like:
```java
public abstract void put(final byte[] key, final byte[] value);
public abstract void put(final ColumnFamilyHandle columnFamilyHandle,
      final byte[] key, final byte[] value) throws RocksDBException;
```

The java developer can then throw an exception in `put` to return a non-OK status.

I think the other functions, `merge`, `delete` and `deleteRange` should also follow the above pattern.

Reply:
Hi, @adamretter, just saw this as for some reason the email notifications went to the junk folder. Please feel free to take over the changes, as I am not active on this at the moment